spring-cloud-aws-autoconfigure/src/main/java/io/awspring/cloud/autoconfigure/dynamodb/DynamoDbEnhancedClientAutoConfiguration.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2013-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.awspring.cloud.autoconfigure.dynamodb;
+
+import org.springframework.boot.autoconfigure.AutoConfiguration;
+import org.springframework.boot.autoconfigure.AutoConfigureAfter;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.annotation.Bean;
+import software.amazon.awssdk.enhanced.dynamodb.DynamoDbEnhancedAsyncClient;
+import software.amazon.awssdk.enhanced.dynamodb.DynamoDbEnhancedClient;
+import software.amazon.awssdk.services.dynamodb.DynamoDbAsyncClient;
+import software.amazon.awssdk.services.dynamodb.DynamoDbClient;
+
+/**
+ * {@link DynamoDbEnhancedClientAutoConfiguration Auto-configuration} for DynamoDB integration.
+ *
+ * @author Matej Nedic
+ * @author Arun Patra
+ * @since 3.0.0
+ */
+@AutoConfiguration
+@ConditionalOnClass({DynamoDbEnhancedClient.class, DynamoDbEnhancedAsyncClient.class})
+@AutoConfigureAfter(DynamoDbClientAutoConfiguration.class)
+@ConditionalOnProperty(name = "spring.cloud.aws.dynamodb.enabled", havingValue = "true", matchIfMissing = true)
+public class DynamoDbEnhancedClientAutoConfiguration {
```

Comment:
Move this class as static inner class of `DynamoDbClientAutoConfiguration` and have `@ConditionalOnClass({DynamoDbEnhancedClient.class, DynamoDbEnhancedAsyncClient.class})`
on top together with `@Configuration`